llm-chain-local/src/options.rs

```diff
@@ -0,0 +1,87 @@
+use llm::{InferenceParameters, InferenceWithPromptParameters, ModelParameters};
+use llm_chain::traits::Options;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+/// An overridable collection of configuration parameters for an LLM. It is combined with a prompt to create an invocation.
+pub struct PerInvocation {
+    pub n_threads: Option<usize>,
+    pub n_tok_predict: Option<usize>,
+    pub top_k: Option<usize>,
+    pub top_p: Option<f32>,
+    pub temp: Option<f32>,
+    pub repeat_penalty: Option<f32>,
+    pub stop_sequence: Option<String>,
+}
+
+impl Options for PerInvocation {}
+
+impl Into<ModelParameters> for PerInvocation {
+    fn into(self) -> ModelParameters {
+        let inference_params = InferenceParameters {
+            n_threads: self.n_threads.unwrap_or(4),
+            n_batch: 8,
+            top_k: self.top_k.unwrap_or(40),
+            top_p: self.top_p.unwrap_or(0.95),
+            repeat_penalty: self.temp.unwrap_or(1.3),
+            temperature: self.temp.unwrap_or(0.8),
+            bias_tokens: Default::default(),
+        };
+
+        let prompt_params = InferenceWithPromptParameters {
+            play_back_previous_tokens: false,
+            maximum_token_count: None,
+        };
+
+        ModelParameters {
+            prefer_mmap: true,
+            n_context_tokens: 2048,
+            inference_params,
+            inference_prompt_params: prompt_params,
+        }
+    }
+}
+
+/// `PerExecutor` represents a collection of configuration parameters for the executor of the LLM.
+/// It contains optional fields for the model path and context parameters.
+///
+/// # Examples
+///
+/// ```
+/// use llm_chain_local::PerExecutor;
+/// let executor = PerExecutor::new().with_model_path("path/to/model");
+/// ```
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct PerExecutor {
+    /// Optional path to the LLM.
+    pub model_path: Option<String>,
+    /// Optional type (e.g. LLaMA, GPT-Neo-X) of the LLM.
+    pub model_type: Option<String>,
```

Comment:
No ideas to have an enum and a fallback? To let know what models are supported?

I mean something like:
`enum ModelType {
  LLama,
  GPT-Neo-X
  Custom(String)
  }
 `

Reply:
The model architecture enum isn't currently serializable (although I don't see why it can't be), so I'm leaving this a string for now.

Reply:
Ah alright! I see!

Reply:
Hopefully this will be changed soon https://github.com/rustformers/llm/pull/200